src/Symfony/Component/Console/Color.php

```diff
@@ -164,6 +170,27 @@ private function degradeHexColorToAnsi(int $r, int $g, int $b): int
         return (round($b / 255) << 2) | (round($g / 255) << 1) | round($r / 255);
     }
 
+    /**
+     * Inspired from https://github.com/ajalt/colormath/blob/e464e0da1b014976736cf97250063248fc77b8e7/colormath/src/commonMain/kotlin/com/github/ajalt/colormath/model/Ansi256.kt code (MIT license).
+     */
+    private function degradeHexColorToAnsi8(int $r, int $g, int $b): int
+    {
+        if ($r === $g && $g === $b) {
+            if ($r < 8) {
+                return 16;
+            } elseif ($r > 248) {
+                return 231;
+            } else {
+                return (int) (round((($r - 8) / 247) * 24)) + 232;
+            }
+        } else {
+            return 16 +
```

Comment:
This should stay on 1 line

Reply:
Harder to read and understand: one line per color. Oneline will use 116 characters + indentation (with many logical group), it's a lot.